pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -52,61 +61,53 @@ protected int getReplication(TableConfig tableConfig) {
   @Override
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
+    SegmentAssignmentStrategy segmentAssignmentStrategy = _assignmentStrategyMap.get(InstancePartitionsType.OFFLINE);
+    // Need to check assignmentStrategy for Dim Tables as following pre conditions state would return false
+    if (segmentAssignmentStrategy instanceof DimTableSegmentAssignmentStrategy) {
```

Comment:
I'm still not sure why we need to handle `DimTableSegmentAssignmentStrategy` separately?